### DISCUSSION TITLE
Language detection giving error in live transcription

### DISCUSSION THREAD

**Discussion by theavijitsarkar** [deepgram/community#850] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - STT Streaming

### Details

I am trying to enable language detection like this

  const dgConnection = deepgram.listen.live({
    model: "nova-2-general",
    detect_language: true,
    language: "hi",
    endpointing: true,
    filler_words: false,
  });

But I am getting an error

Error: Unexpected server response: 400
      at ClientRequest.<anonymous> (/Applications/MAMP/htdocs/native-voicebot/node_modules/ws/lib/websocket.js:913:7)
      at ClientRequest.emit (node:events:517:28)
      at HTTPParser.parserOnIncomingClient (node:_http_client:700:27)
      at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
      at TLSSocket.socketOnData (node:_http_client:541:22)
      at TLSSocket.emit (node:events:517:28)
      at addChunk (node:internal/streams/readable:335:12)
      at readableAddChunk (node:internal/streams/readable:308:9)
      at Readable.push (node:internal/streams/readable:245:10)
      at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23),

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

/v1/listen?model=nova-2-general&detect_language=true&language=hi&endpointing=true&filler_words=false

### If possible, please attach your code or paste it into the text box.

  const dgConnection = deepgram.listen.live({
    model: "nova-2-general",
    detect_language: true,
    language: "hi",
    endpointing: true,
    filler_words: false,
  });

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment, marked as answer]
@theavijitsarkar 

Our language detection feature is not available for the live streaming STT API, only for our prerecorded API. You are getting the 400 error because you've turned on language detection for streaming.

**Reply by SandraRodgers** [reply]
@theavijitsarkar See my edited comment above ^^^

**Reply by theavijitsarkar** [reply]
Any update when It will be available? It is required for voice bots, which use streaming to reduce latency

**Reply by SandraRodgers** [reply]
Which languages do you need this for? We've just released this for English/Spanish but I can try to find out if other languages are in the pipeline.  https://developers.deepgram.com/docs/multilingual-code-switching

**Reply by theavijitsarkar** [reply]
Hindi would be critical for our use case, actually eventually all languages.  Ideally, if could remove the language parameter during setup all together so that the live transcription always detects and responds with the language in the Transcript event, this will let us build multilingual bots

**Reply by theavijitsarkar** [reply]
Looks like the multilingual feature is what I am talking about, thanks let me give it a go

**Reply by SandraRodgers** [reply]
We have many requests for this type of feature and the team is definitely working on it. I've passed on your feedback that Hindi would be your highest need. Thank you!

**Reply by olsihoxha** [reply]
@SandraRodgers do you have any information if the Italian language is in the pipeline ?
